### PR TITLE
feat: Updated feast Go operator db stores

### PR DESCRIPTION
**Pull request by tmihalac**
# What this PR does / why we need it:
This PR contains the changes and leftovers from [4771](https://github.com/feast-dev/feast/pull/4771)

# Which issue(s) this PR fixes:
Fixes https://github.com/feast-dev/feast/issues/4793
Fixes https://github.com/feast-dev/feast/issues/4783

### CODE REVIEW

**Review by tchughesiv — APPROVED** [review]
lgtm

File: infra/feast-operator/config/samples/v1alpha1_featurestore_db_persistence.yaml [hunk 1]
```diff
@@ -11,5 +11,25 @@ spec:
...
+kind: Secret
+metadata:
+  name: postgres-secret
+stringData:
+  postgres-secret-parameters: |
+    path: postgresql://postgres:mysecretpassword@127.0.0.1:55001/feast
+    cache_ttl_seconds: 60
+    sqlalchemy_config_kwargs:
+      echo: false
+      pool_pre_ping: true
```
Comment (tchughesiv) [inline infra/feast-operator/config/samples/v1alpha1_featurestore_db_persistence.yaml:35]: this sample wouldn't work without some changes -
```suggestion
            name: postgres-secret
    registry:
      local:
        persistence:
          store:
            type: sql
            secretRef:
              name: postgres-secret
            secretKeyName: postgres-secret-parameters   # optional
---
apiVersion: v1
kind: Secret
metadata:
  name: postgres-secret
stringData:
  postgres-secret-parameters: |
    path: postgresql+postgresql://postgres:mysecretpassword@127.0.0.1:55001/feast
    cache_ttl_seconds: 60
    sqlalchemy_config_kwargs:
      echo: false
      pool_pre_ping: true
  postgres: |
    host: 127.0.0.1
    port: 55001
    database: feast
    db_schema: public
    user: postgres
    password: mysecretpassword
```

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: infra/feast-operator/config/samples/v1alpha1_featurestore_db_persistence.yaml (hunk 1 above)
Comment (tchughesiv) [reply infra/feast-operator/config/samples/v1alpha1_featurestore_db_persistence.yaml:35]: please verify that this sample works in a cluster

File: infra/feast-operator/config/samples/v1alpha1_featurestore_db_persistence.yaml [hunk 2]
```diff
@@ -11,5 +11,32 @@ spec:
...
+              name: postgres-secret
+            secretKeyName: postgres-secret-parameters   # optional
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: postgres-secret
+stringData:
+  postgres-secret-parameters: |
+    path: postgresql+postgresql://postgres:mysecretpassword@127.0.0.1:55001/feast
```
Comment (tchughesiv) [inline infra/feast-operator/config/samples/v1alpha1_featurestore_db_persistence.yaml:31]:
```suggestion
    path: postgresql+psycopg://postgres:mysecretpassword@127.0.0.1:55001/feast
```

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: infra/feast-operator/config/samples/v1alpha1_featurestore_db_persistence.yaml [hunk 3]
```diff
@@ -11,5 +11,32 @@ spec:
...
+            name: postgres-secret
+          secretKeyName: postgres-secret-parameters   # optional
+    registry:
+      local:
+        persistence:
+          store:
+            type: sql
+            secretRef:
+              name: postgres-secret
+            secretKeyName: postgres-secret-parameters   # optional
```
Comment (tchughesiv) [inline infra/feast-operator/config/samples/v1alpha1_featurestore_db_persistence.yaml:23]:
```suggestion
            secretKeyName: postgres-secret-parameters   # optional, will use store.type by default... in this case that's "sql"
```

**Review by tchughesiv — CHANGES_REQUESTED** [review]

**Review by franciscojavierarceo — APPROVED** [review]

**Pull request merged**